packages/components/src/public/QueryInfo.ts

```diff
@@ -258,10 +259,31 @@ export class QueryInfo {
         return extraDisplayColumn;
     }
 
-    getLookupViewColumns(omittedColumns?: string[]): QueryColumn[] {
-        const lcCols = omittedColumns ? omittedColumns.map(c => c.toLowerCase()) : [];
-        return this.columns.filter(col => col.shownInLookupView && lcCols.indexOf(col.fieldKey.toLowerCase()) === -1)
-            .valueArray;
+    getLookupViewColumns(displayColumnFieldKey?: string): QueryColumn[] {
+        let cols: QueryColumn[] = [];
+        if (this.views.has(IDENTIFYING_FIELDS_VIEW_NAME)) {
+            this.views.get(IDENTIFYING_FIELDS_VIEW_NAME).columns.forEach(col => {
+                const qCol = this.columns.get(col.fieldKey.toLowerCase());
```

Comment:
nit: Prefer `const qCol = this.getColumn(col.fieldKey);`

Reply:
Done.